libs/ui/src/precinct_scanner_full_report.tsx

```diff
@@ -0,0 +1,81 @@
+import {
+  ElectionDefinition,
+  getPartyIdsInBallotStyles,
+  PrecinctSelection,
+  Tally,
+} from '@votingworks/types';
+import { assert, getTallyIdentifier } from '@votingworks/utils';
+import React from 'react';
+import { PrecinctScannerTallyQrCode } from './precinct_scanner_tally_qrcode';
+import { PrecinctScannerTallyReport } from './precinct_scanner_tally_report';
+
+export interface PrecinctScannerFullReportProps {
+  electionDefinition: ElectionDefinition;
+  precinctSelectionList: PrecinctSelection[];
+  subTallies: ReadonlyMap<string, Tally>;
```

Comment:
This is not the best or safest data type to pass this type of data around, especially because we're keying it with values from `getTallyIdentifier` that could be problematic later on. Since the current logic (seems to me) is based on the possibility that we are unable to save precinct tallies to the card for lack of space, and that limitation will likely lift with java cards, I don't want to refactor much here.